src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -138,6 +144,58 @@ public void End(ProduceEndType endType)
             }
         }
 
+        public MemoryPoolIterator2 ProducingStart()
+        {
+            lock (_returnLock)
+            {
+                Debug.Assert(!_isProducing);
+                _isProducing = true;
+
+                if (_tail == null)
+                {
+                    throw new IOException("The socket has been closed.");
+                }
+
+                return new MemoryPoolIterator2(_tail, _tail.End);
```

Comment:
Shouldn't this behave like the `StreamSocketOutput.ProducingStart()`; use no lock and return a `MemoryPoolIterator2` with a fresh LeasedBlock so its disconnected? Rather than essentially returning the active queue.